pkg/auth/auth.go

```diff
@@ -139,16 +105,10 @@ func redactClientCredentials(sensitiveString string) string {
 	return r.ReplaceAllString(sensitiveString, "$1##### REDACTED #####$3")
 }
 
-// addTargetTypeHeader adds the target header if proxy mode is enabled.
-func addTargetTypeHeader(spt *adal.ServicePrincipalToken) *adal.ServicePrincipalToken {
-	spt.SetSender(autorest.CreateSender(
-		(func() autorest.SendDecorator {
-			return func(s autorest.Sender) autorest.Sender {
-				return autorest.SenderFunc(func(r *http.Request) (*http.Response, error) {
-					r.Header.Set(consts.RequestHeaderTargetType, consts.TargetTypeAzureActiveDirectory)
-					return s.Do(r)
-				})
-			}
-		})()))
-	return spt
+type transporter struct{}
+
+func (t *transporter) Do(req *http.Request) (*http.Response, error) {
+	// adds the target header if proxy mode is enabled
+	req.Header.Set(consts.RequestHeaderTargetType, consts.TargetTypeAzureActiveDirectory)
+	return http.DefaultClient.Do(req)
```

Comment:
Using the default client seems a bit sketch?

Reply:
@aramase :grimacing: